### PR TITLE
Add a new crate for claims

### LINKED ISSUE #4196: Add structs for working with claims and in-toto statements
We need these strcuts both in the ML-transparency runner, and for parsing and verifying endorsement statements.

### PR DESCRIPTION
**Pull request by rbehjati**
Fixes #4196.

- Also fixes issues with the dependencies on bitflags. 
- Updated oak_ml_transparency to use the claim structs. Builds a new release of the runner, and updates the Dockerfile and the Github workflow file to use the latest runner.

TODO in a future PR: 
- This PR adds a new endorsement.json file for testing. This should be merged with the existing one. I'll do that once https://github.com/project-oak/oak/pull/4206 is merged.

### CODE REVIEW

File: java/src/main/java/com/google/oak/transport/GrpcStreamingTransport.java [hunk 1]
```diff
@@ -67,8 +65,8 @@ public GrpcStreamingTransport(
   @Override
   public Result<AttestationBundle, String> getEvidence() {
     RequestWrapper requestWrapper = RequestWrapper.newBuilder()
-        .setGetPublicKeyRequest(GetPublicKeyRequest.newBuilder())
-        .build();
+                                        .setGetPublicKeyRequest(GetPublicKeyRequest.newBuilder())
```
Comment (conradgrobler) [inline java/src/main/java/com/google/oak/transport/GrpcStreamingTransport.java:68]: I don't think this reformatting should be part of this PR. If the current java formatting is incorrect (or if our formatting script uses the wrong format) that should be solved in a different PR.

File: Cargo.lock [hunk 2]
```diff
@@ -2241,7 +2259,7 @@ dependencies = [
 name = "oak_stage0"
 version = "0.1.0"
 dependencies = [
- "bitflags 1.3.2",
+ "bitflags 2.3.3",
```
Comment (conradgrobler) [inline Cargo.lock:2262]: This change seems problematic (also the `oak_linux_boot_params` bitflags dependency above). Rather than having to fix the bitflags version in every cargo.toml, perhaps it could be fixed by using `cargo update` with `--precise` to force a specific version for bitflags on the entire workspace.

**Review by conradgrobler — APPROVED** [review]

**Review by rbehjati — COMMENTED** [review]
Thanks for the review.

File: Cargo.lock (hunk 2 above)
Comment (rbehjati) [reply Cargo.lock:2262]: Also, it does not quite work:

```
$ cargo update -p bitflags@2.3.3 --precise 1.3.2
    Updating crates.io index
error: failed to select a version for the requirement `bitflags = "^2.3.0"`
candidate versions found which didn't match: 1.3.2
location searched: crates.io index
required by package `virtio-drivers v0.5.0`
    ... which satisfies dependency `virtio-drivers = "*"` (locked to 0.5.0) of package `oak_restricted_kernel v0.1.0 (/usr/local/google/home/razieh/dev/code/project-oak/oak/oak_restricted_kernel)`
perhaps a crate was updated and forgotten to be re-vendored?
```

File: claims/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+[package]
+name = "claims"
```
Comment (tiziano88) [inline claims/Cargo.toml:2]: Remember (from the other conversations) that crates live in a global flat namespace, so they need to have globally unique names. In particular, it looks like this name is in fact already taken: https://crates.io/crates/claims . This caused us actual issues in the past, in which cargo would sometimes start depending on the external crate without a way of overriding it.

Please use a name like oak_transparent_claims or at least oak_claims

File: claims/src/claims.rs [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+/// URI representing the PredicateType of a V1 Claim. To be used in in-toto statements.
+pub const CLAIM_V1: &str = "https://github.com/project-oak/transparent-release/claim/v1";
+
+/// ClaimType for Endorsements. This is expected to be used together with `ClaimV1` as the predicate
+/// type in an in-toto statement.
+pub const ENDORSEMENT_V2: &str =
+    "https://github.com/project-oak/transparent-release/endorsement/v2";
+
+/// Trait representing a claim specification.
+pub trait ClaimSpec {}
```
Comment (tiziano88) [inline claims/src/claims.rs:36]: Not sure I understand the point of this empty trait, could you add a comment? If it's a marker trait, please make that explicit. The fact that it's public though does not really make it useful, since anyone can trivially implement it on any type.

File: claims/src/claims.rs [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+pub enum InvalidClaimData {
+    StatementType,
+    PredicateType,
+    ClaimType,
+    Validity(String),
+}
+
+/// Detailed content of a claim.
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+pub struct ClaimPredicate<S: ClaimSpec> {
```
Comment (tiziano88) [inline claims/src/claims.rs:47]: Does this need to be kept in sync with some other schema? If so, could you link to the canonical reference for it?

File: oak_containers_kernel/configs/6.1.33/minimal.config [hunk 6]
```diff
@@ -11,7 +11,6 @@ CONFIG_AS_VERSION=24000
 CONFIG_LD_IS_BFD=y
 CONFIG_LD_VERSION=24000
 CONFIG_LLD_VERSION=0
-CONFIG_RUST_IS_AVAILABLE=y
```
Comment (tiziano88) [inline oak_containers_kernel/configs/6.1.33/minimal.config:14]: Is this intentional?

File: oak_ml_transparency/runner/Cargo.toml [hunk 7]
```diff
@@ -10,10 +10,12 @@ license = "Apache-2.0"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 [dependencies]
 anyhow = "*"
+claims = { path = "../../claims" }
```
Comment (tiziano88) [inline oak_ml_transparency/runner/Cargo.toml:13]: I think this should just be marked as a workspace-level dependency, since you already added it there.

File: claims/Cargo.toml (hunk 3 above)
Comment (ipetr0v) [reply claims/Cargo.toml:2]: Or `oak_transparency_claims`

File: oak_containers_kernel/configs/6.1.33/minimal.config (hunk 6 above)
Comment (ipetr0v) [reply oak_containers_kernel/configs/6.1.33/minimal.config:14]: This happens after `xtask format`

I had to revert it in my PR once

**Review by ipetr0v — APPROVED** [review]

File: claims/Cargo.toml (hunk 3 above)
Comment (rbehjati) [reply claims/Cargo.toml]: Changed the crate name to `oak_transparency_claims`.

File: claims/src/claims.rs (hunk 4 above)
Comment (rbehjati) [reply claims/src/claims.rs]: Good point. Removed it.

File: claims/src/claims.rs (hunk 5 above)
Comment (rbehjati) [reply claims/src/claims.rs]: Added a comment about it.

File: oak_containers_kernel/configs/6.1.33/minimal.config (hunk 6 above)
Comment (rbehjati) [reply oak_containers_kernel/configs/6.1.33/minimal.config]: For me running cargo test removed it! Fixed.

File: oak_ml_transparency/runner/Cargo.toml (hunk 7 above)
Comment (rbehjati) [reply oak_ml_transparency/runner/Cargo.toml]: oak_ml_transparency/runner/Cargo.toml in not in the top-level workspace. It is a separate one.

File: Cargo.lock (hunk 2 above)
Comment (rbehjati) [reply Cargo.lock]: Created to https://github.com/project-oak/oak/issues/4223, to track fixing the issue the right way.

**Review by rbehjati — COMMENTED** [review]
Thanks for the reviews. 

Pinning the version is a temporary solution. Created #4223 to make sure we'll remember to fix the issue the right way.

File: oak_simple_io/Cargo.toml [hunk 8]
```diff
@@ -6,6 +6,5 @@ edition = "2021"
 license = "Apache-2.0"
 
 [dependencies]
-bitflags = "*"
```
[the remark this answers is not in the document]
Comment (conradgrobler) [reply oak_simple_io/Cargo.toml:9]: If everything still builds without it I guess it was no longer used.

**Review by tiziano88 — APPROVED** [review]
